imagetest/test_suites/image_validation/license_test.go

```diff
@@ -0,0 +1,203 @@
+	`is (available|distributed) under the terms of the GNU.*Public License`,
+	`(libudev|libgudev|udev) is licensed under the GNU (L|)GPL`,
+	`The Linux Console Tools are covered by the GPL`,
+	`Some portions of os-prober`,
+	`Netcat and the associated package is a product of Avian Research, and is freely available in full source form with no restrictions save an obligation to give credit where due.`,
+	`Permission is hereby granted, without written agreement and without licence or royalty fees, to use, copy, modify, and distribute this software`,
+	`Open Market permits you to use, copy, modify, distribute, and license this Software and the Documentation for any purpose, provided that existing copyrightPathGlob notices are retained in all copies and that this notice is included verbatim in any distributions. No written agreement, license, or royalty fee is required for any of the authorized uses.`,
+	`This software is made available under the terms of *either* of the licenses found in LICENSE.APACHE or LICENSE.BSD. Contributions to cryptography are made under the terms of *both* these licenses.`,
+}
+
+const (
+	copyrightPathGlob = "/usr/share/doc/*/copyrightPathGlob"
+	licensePathGlob   = "/usr/share/doc/*/LICENSE"
+	licenseNameRegex  = `(?i)((?:(?:License|Copyright)\s*:\s*%[1]s)|(?:(?:covered )*under (?:the )?%[1]s)|(?:under (?:the terms of )*the %[1]s))`
+)
+
+func isValidLicenseName(licenseCheck string) bool {
+	for _, name := range licenseNames {
+		var regexString = fmt.Sprintf(licenseNameRegex, name)
+		re := regexp.MustCompile(regexString)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func isValidLicenseText(licenseCheck string) bool {
+	for _, licenseText := range licenses {
+		// (?i) case insensitive
+		re := regexp.MustCompile(`(?i)` + licenseText)
+
+		if re.MatchString(licenseCheck) {
+			return true
+		}
+	}
+	return false
+}
+
+func TestArePackagesLegal(t *testing.T) {
+	var pathGlob string
+
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+
+	switch {
+	case strings.Contains(image, "rhel"), strings.Contains(image, "suse"):
+		pathGlob = licensePathGlob
+	default:
+		pathGlob = copyrightPathGlob
+	}
+	filenames, err := filepath.Glob(pathGlob)
+	if err != nil {
+		t.Fatalf("couldnt resolve glob %s", pathGlob)
+	}
+
+	for _, filename := range filenames {
+		isLagel, err := isPackageLegal(filename)
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+		if !isLagel {
+			t.Fatalf("Found illegal package: %v", filename)
+		}
+	}
+}
+
+func isPackageLegal(filepath string) (bool, error) {
+	bytes, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return false, fmt.Errorf("error read file")
+	}
+
+	var licenseCheck string = string(bytes)
+
+	// Remove comment
+	re := regexp.MustCompile(`(\*|#).`)
+	re.ReplaceAllString(licenseCheck, "")
+
+	// Replace all whitespace with one space
+	whitespaceRegex := regexp.MustCompile(`\s+`)
```

Comment:
where is this in the original python code? i think that code joins lines, rather than replaces whitespace. regular expressions don't span lines

Reply:
new code should be:
```
licenseCheck = strings.ReplaceAll(licenseCheck, "\n", " ")
```
to match original code:
```
license_check = ' '.join(license_check.split())
```

and remove whitespaceregex

Reply:
what do you suggest for fix? what's missing in the whitespaceRegex is newlines. we must strip newlines or the regex will often fail in this new code where it would have succeeded in the old